integration_tests/run_pyspark_from_build.sh

```diff
@@ -26,8 +26,14 @@ then
 else
     echo "WILL RUN TESTS WITH SPARK_HOME: ${SPARK_HOME}"
     [[ ! -x "$(command -v zip)" ]] && { echo "fail to find zip command in $PATH"; exit 1; }
-    VERSION_STRING=`$SPARK_HOME/bin/pyspark --version 2>&1|grep -v Scala|awk '/version\ [0-9.]+/{print $NF}'`
-    VERSION_STRING="${VERSION_STRING/-SNAPSHOT/}"
+    PY4J_TMP=("${SPARK_HOME}"/python/lib/py4j-*-src.zip)
+    PY4J_FILE=${PY4J_TMP[0]}
+    # PySpark uses ".dev0" for "-SNAPSHOT",  ".dev" for "preview"
```

Comment:
Have we verified that Cloudera and other non-standard Spark versioning schemes work with this change?

Reply:
only on Databricks

Reply:
Judging by #5146 we use `INTEGRATION_TEST_VERSION_OVERRIDE` for Cloudera

I assume it's similarly overridden for EMR because the current code does not handle it correctly:

```bash
$ /usr/lib/spark/bin/pyspark --version 2>&1 | grep -v Scala | awk '/version\ [0-9.]+/{print $NF}'
3.3.2-amzn-0
```

Updated code results in a similar string with '-' replaced by '+'
```bash
$ export SPARK_HOME=/usr/lib/spark
$ PY4J_TMP=("${SPARK_HOME}"/python/lib/py4j-*-src.zip)
$ PYTHONPATH=${SPARK_HOME}/python:${PY4J_FILE} python -c "import pyspark, re; print(re.sub('\.dev0?$', '', pyspark.__version__))"
3.3.2+amzn.0
```

Reply:
Then it seems like this change is no worse than what we were already doing.  Thanks for digging, @gerashegalov!

Reply:
@pxLi @zhanga5 can you confirm that we don't need to worry about distros with modified versions   `INTEGRATION_TEST_VERSION_OVERRIDE` for non-standard builds?